Added Translation support and Spanish translation.

EdgarJRobles opened the pull request:
Menu bar were translated.
object name were translated.

EdgarJRobles committed:
Added translation support

Added Pyside2 support translate dodo external workbench from FreeCAD.

Added Spanish translation.

Added TS and QM

spanish translation files

Added headers and more translation functions

Added QT_TRANSLATE_NOOP definition

Added addproperty translation

Minor  addproperty translations

Minor  addproperty translations

Adde more translated lines

Addobject name translated
Release last revision es-ES QM file

Fixed typo on addproperty

Wrong typo syntax

Added locale update function

Added locale update function

Fixed addObject translation / Erased context in ts file

-Fixed addObject translation.
-Erased not used context formats in TS file.

Translate Drop down headers

Translate drop down headers

Minor accent correction

EdgarJRobles committed:
Added translation support

Added Pyside2 support translate dodo external workbench from FreeCAD.

Added Spanish translation.

Added TS and QM

spanish translation files

Added headers and more translation functions

Added QT_TRANSLATE_NOOP definition

Added addproperty translation

Minor  addproperty translations

Minor  addproperty translations

Adde more translated lines

Addobject name translated
Release last revision es-ES QM file

Fixed typo on addproperty

Wrong typo syntax

Added locale update function

Added locale update function

Fixed addObject translation / Erased context in ts file

-Fixed addObject translation.
-Erased not used context formats in TS file.

Translate Drop down headers

Translate drop down headers

Minor accent correction

EdgarJRobles committed:
rebase Minor  addproperty translations

EdgarJRobles committed:
rebase Minor  addproperty translations

EdgarJRobles committed:
merge pCmd with translated function
Addobject name translated
Release last revision es-ES QM file

EdgarJRobles committed:
merge pCmd with translated function
Addobject name translated
Release last revision es-ES QM file

EdgarJRobles committed:
Fixed typo on addproperty

Wrong typo syntax

EdgarJRobles committed:
Fixed typo on addproperty

Wrong typo syntax

EdgarJRobles committed:
Added locale update function

Added locale update function

EdgarJRobles committed:
Added locale update function

Added locale update function

luzpaz commented:
@EdgarJRobles does the translation work correctly ?

luzpaz commented on translations/Update code.txt:
```diff
@@ -0,0 +1 @@
+pylupdate5 ./../InitGui.py ./../fCmd.py ./../CPipe.py ./../CUtils.py ./../dodoDialogs.py ./../dodoPM.py ./../fForms.py ./../fObservers.py ./../CFrame.py ./../pFeatures.py ./../pCmd.py  -ts ./dodo_es-ES.ts
```
`Update code.txt` ?  
@EdgarJRobles Is this file name auto-generated or what you named it?

EdgarJRobles replied:
It's a comamnd that use to upgrade and generate TS file from python files.

EdgarJRobles replied:
It comes from Pyqt5

EdgarJRobles replied:
> `Update code.txt` ?
> @EdgarJRobles Is this file name auto-generated or what you named it?

It is not auto-generated, i do it manually

FilePhil committed:
Added Cutlist Feature Folder

FilePhil committed:
Added Cut List Creation to Frame Tools

FilePhil committed:
changed Command Name to fit in with the workbench naming Schema

FilePhil committed:
Added GitIgnore

FilePhil committed:
Added Readme to Cutlist Module

FilePhil committed:
Changed Readme from Macro Version to integrated Version

hasecilu committed:
Add bash script to update translations

hasecilu committed:
Add bash script to update translations

hasecilu committed:
Add bash script to update translations

hasecilu committed:
Add bash script to update translations

hasecilu committed:
Rename classes' names

It's needed for at the time of extracting the strings for translation
the strings are separated by context making easier translation process.

hasecilu committed:
Rename classes' names

It's needed for at the time of extracting the strings for translation
the strings are separated by context making easier translation process.

EdgarJRobles commented:
I can confirm the translation works on FreeCAD 0.22.0, Libs: 0.22.0devR36338
![Screenshot_20240313_214213](https://github.com/oddtopus/dodo/assets/13512639/21af5ffb-481a-4b5a-8d49-110d519e7140)
![Screenshot_20240313_215658](https://github.com/oddtopus/dodo/assets/13512639/bc54850b-300b-4677-81b9-e494b5218b2f)

hasecilu commented:
Same here, translation working as expected, there are still some strings on dialogs missing like `self.labName = QtGui.QLabel("(seleziona un oggetto)", self)`.

@oddtopus We would love your feedback to make translation support happen.

hasecilu committed:
Mark missing strings for translation

hasecilu committed:
WIP: Mark missing strings for translation

- makeHeader() MenuText & ToolTip
- Dialogs
- Transactions (undo/redo)
- Tree View Labels
- Observers messages

hasecilu committed:
Mark missing strings for translation

hasecilu committed:
WIP: Mark missing strings for translation

- makeHeader() MenuText & ToolTip
- Dialogs
- Transactions (undo/redo)
- Tree View Labels
- Observers messages

hasecilu committed:
Update Spanish translation

hasecilu committed:
Update Spanish translation

hasecilu commented:
STATS:
- 49 `PrintMessage()`
- 19 `PrintWarning()`
- 55 `PrintError()`
- 51 `openTransaction()`
- 129 `setText()`

EdgarJRobles committed:
Mayor extra translations from code to ts file

EdgarJRobles committed:
Mayor extra translations from code to ts file

EdgarJRobles committed:
remove older translates folder & fix conflicts with pFeatures.py

EdgarJRobles committed:
remove older translates folder & fix conflicts with pFeatures.py

EdgarJRobles committed:
Spanish translation complete

EdgarJRobles committed:
Spanish translation complete

FilePhil committed:
fixed Dataclass List for Python Version <3.9

EdgarJRobles committed:
Add FreeCAD 1.1.0 support on fFeatures.py & normalize language path in pForms.py

EdgarJRobles committed:
Add FreeCAD 1.1.0 support on fFeatures.py & normalize language path in pForms.py

hasecilu committed:
Fix `.gitignore`

hasecilu committed:
Update translations script

The previous way on which the files were updated is done is by creating
some temporal files and then merging them to the original file.

Doing this merge would let us to NOT lost the translations already done.

It was done like that because we got the string from Python files and UI
files because `lupdate` from Qt5 couldn't extract strings from Python
files.

Now the `lupdate` from Qt6 can do, so we have no need to merge different
temp files, instead, we can do the update in a single command.

The new advantage is that the `<location>` tag is updated with all new
occurrences.

hasecilu committed:
Add `translationz/README.md` file

hasecilu committed:
fixup! Rename classes' names

hasecilu committed:
Update README file

hasecilu committed:
Add ruff config file to format Python files

Intended to be used in conjunction with Neovim & LazyVim.
The default LazyVim's Python specification use ruff-lsp to format files,
this file let you configure ruff, mainly, the line width to 100 chars.

hasecilu committed:
Add pre-commit hooks

Using the ones are used on main FreeCAD plus codespell

hasecilu committed:
Update `package.xml` file

ebrahimraeyat committed:
ChangeProfile don't remove sketch profile

From https://github.com/oddtopus/dodo/pull/9
e9b4201fe18a81a107ba5baba9f8777472aea78c

realthunder committed:
Fix touched object after recomputation

From https://github.com/oddtopus/dodo/pull/19/files
a6978ccda2ec993120a09d2fb9810924661acfbd

chennes committed:
Make license match SPDX ID

From https://github.com/oddtopus/dodo/pull/24
7888ed2812b2d4bccd813972387d5565741ed787

EdgarJRobles committed:
CutList integration, translation & new mitermethod

-Fixed duplicated utils toolbar
-Cut list dialog incluyed into bash script translation & translated to spanish
-Incluye parcial method to generate bisect planes in sketckes vertexes

hasecilu committed:
Fix last name

hasecilu committed:
Rename commands to have WB prefix

Needed to properly create Wiki articles for documentation and make
translations work properly.

- Fix contexts on commands' MenuText and ToolTip strings

In order to have the translation working using the QT_TRANSLATE_NOOP()
function on the commands the context must match the command name and not
the name of the class that is executed.

Example:
$ sed -i 's/\"cutListCommand/\"Dodo_CreateCutList/g' cut_list/*.py

hasecilu committed:
Update translations file

hasecilu committed:
Fix typos

hasecilu committed:
Create utility file to get paths

hasecilu committed:
Fix contexts of properties strings

Regarding of the type of the property the context always is:
"App::Property"

EdgarJRobles committed:
Createcutlist to toolist & package.xml minor update

EdgarJRobles committed:
Update README.md

hasecilu committed:
Update URL on badges

EdgarJRobles committed:
remove pyrightconfig.json from repository on gitignore

EdgarJRobles committed:
Merge branch 'feat/translation++' of github.com:EdgarJRobles/dodo into feat/translation++

EdgarJRobles committed:
extend RH profile csv file

hasecilu committed:
Remove print statements

hasecilu committed:
Rename workbench

hasecilu committed:
Don't translate objects' internal name

The procedure for translation should be to translate the string that is
assigned to the label: `obj.Label`

hasecilu committed:
Make FreeCAD version check work with 1.0 version

Instead of making a comparison with int variable, a comparison with
float works fine.

hasecilu committed:
Use translate functions from `FreeCAD.Qt`, reorganize imports

Used "code actions"

hasecilu committed:
Fix typos

hasecilu committed:
Use new function to get icon path

hasecilu committed:
Use ViewProvider to assign icon to parametric objects

hasecilu committed:
Rename QM commands (pie menu)

hasecilu committed:
Update some icons

Get better contrast on dark themes

hasecilu committed:
Mark commands unavailable when there is no active document

EdgarJRobles committed:
Fixed PypeLineForm pipe generation & python files
-Match method & classes spacing
-Include exception condition for OCC Part on pypeline class
-Remove .ts code with no lines asociated

EdgarJRobles committed:
Added miter corner feature & carry returns
+carry return on dodo console messages
+new miter feature automatically remove a diagonal volume

hasecilu committed:
Bump version of pre-commit hooks

hasecilu committed:
Apply "black", 100 char line width style to Python files

Used pre-commit

hasecilu committed:
Update WB icon with an actual quetzal*

* attempt of quetzal, temporarely

hasecilu committed:
Move `addCommand()` function to quetzal_config module

hasecilu committed:
Add templates for bug and feature-request actions

hasecilu committed:
Add `CHANGELOG.md` file

hasecilu committed:
Update some icons, part 2

Get better contrast on dark themes

EdgarJRobles committed:
Replace original profiles to use BIM profiles

EdgarJRobles committed:
Added ASME flange weld neck, lap join, socket slot

-Added round Beam profiles support
-Added extra fillet parameters to apply fillet on flanges
-Added piping flange location based on table flange type

EdgarJRobles committed:
Update  README.md

Update due main folder rename

EdgarJRobles committed:
Merge branch 'feat/BIM_profiles'

hasecilu committed:
Fix typos

hasecilu committed:
Update translations

- Include script to download translations via CrowdIn API
- Update bash script
- Bring new translations from CrowdIn

hasecilu committed:
Add `.editorconfig` file for shell script

hasecilu committed:
Update `package.xml` file

Fix #1

hasecilu committed:
More black formatting to missing files with ruff hook

To have same style on the WB

hasecilu committed:
Mark `openTransaction()` strings for translation

hasecilu committed:
Replace black hook with compatible ruff hook

hasecilu committed:
Use `isinstance()` instead of `==`

EdgarJRobles committed:
Fix object bases & added extra flange & pipe standarts

-Fix attachment support from pipe features
-Added Slip on flanges from ASME standart
-Added extra ASME pipe grades from 10 to 160
-Added tooltip for nozzle tank insert function due unknow behavior

hasecilu committed:
Remove extra imports and duplicated keys

hasecilu committed:
Update some icons, part 3

hasecilu committed:
Use PySide instead of PySide2

hasecilu committed:
Normalize tooltips

hasecilu committed:
Don't translate objects' internal name, part 2

Only translate the strings attached to `obj.Label`

hasecilu committed:
Update some icons, part 4

Some icons were taken from:
https://forum.freecad.org/viewtopic.php?f=34&t=55223
and reworked a little bit.

hasecilu committed:
Rename icons and add them to FreeCAD resources path

- Move icons to FreeCAD's icon list to make them available for PieMenu
- Rename icons to prevent clash with other Wb's icons
- Some commands were renamed to be more explicit on what they do

hasecilu committed:
Clean InitGui

Remove typos from user-facing strings

hasecilu committed:
Update README

hasecilu committed:
Update translationz/README

hasecilu committed:
Add CONTRIBUTING

hasecilu committed:
Update CHANGELOG

hasecilu committed:
Fix declared repository branch

EdgarJRobles committed:
Merge pull request #7 from EdgarJRobles/chore/110

Preparations to 1.1.0

EdgarJRobles committed:
Disable command & extra flange offset over pipes

fix: frame & pipe button always activated
feat: Added checkbox option on flange form to cut pipe based on flange size
feat: Added a combobox to select standarts, still on development

EdgarJRobles committed:
New feature tooltips details & fix open transaction

feat: Added tooltips file explaining tool operation
fix: removed translation on open transaction function due unexpected crash
feat: Added doxygen documentation support

EdgarJRobles committed:
fix flange visual deviation during creation

EdgarJRobles committed:
Add Task panel option & snaptoolbar to Quetzal

feat: Added quetzal task panel
feat: Added draftsnap toolbar in quetzal workbench
fix: Class changed from line to wire used to make lines from draft

EdgarJRobles committed:
fix: translation conflit on Cpipe.py file

EdgarJRobles committed:
Added terminal adapter objects & some fixes

feat(build): add terminal adapter object & EMT, IMT & RMT grade pipe
feat(build): add GUI for terminal adapter object selection
feat(build): add terminal adapter tooltips
feat(build): add terminal adapter iconz
feat(build): add makepolygon function
fix(build): rating does not assign on each pipe with Quetzal_InsertPipe
or Quetzal_Point2Point
fix(build): numerical input does not create pipe with Quetzal_Point2Point function

EdgarJRobles committed:
Add taskwatcher for wire & pipes

feat: taskwatcher for pipe & wire & empty folder
fix: transaction translation fix from fCmd on CPipe
fix: Prating does not assigned

EdgarJRobles committed:
Fix deprecation warning from addextension

fix: Version statement with wrong number

EdgarJRobles committed:
Update README.md

Added piping example

EdgarJRobles committed:
Update README.md

Add just Freecad windows piping example

Syres916 committed:
Prepare for QCheckBox stateChanged -> checkStateChanged

EdgarJRobles committed:
Merge pull request #17 from Syres916/patch-2

Prepare for QCheckBox stateChanged -> checkStateChanged

EdgarJRobles committed:
Add hints
refactor: restructure python hints

Syres916 committed:
tooltips => Quetzal_tooltips

Syres916 committed:
tooltips => Quetzal_tooltips

Syres916 committed:
Create Quetzal_tooltips.py

Syres916 committed:
Delete tooltips.py

EdgarJRobles committed:
Merge pull request #18 from Syres916/patch-3

Change tooltips -> Quetzal_tooltips ensuring there is no clash with…

EdgarJRobles committed:
Dependency issues
fix(config): old Arch dependency remove
fix(config): add typing_extensions module dependency

EdgarJRobles committed:
Dependency issues 2

fix(config): retype typing-extensions module dependency

chennes committed:
Update package.xml dependencies to be more explicit

EdgarJRobles committed:
Merge pull request #21 from chennes/patch-2

Update package.xml dependencies to be more explicit

PhoneDroid committed:
Add SPDX License Identifiers

PhoneDroid committed:
Add previous authors

EdgarJRobles committed:
Merge pull request #23 from PhoneDroid/Add-SPDX

SPDX License Identifiers

EdgarJRobles committed:
fix(typing): Overall length parameter

EdgarJRobles committed:
refactor(ui): simplify framemanager

add button tooltips
add preview profile images

fix(miters):
add TSLOT on do profile method
plane wrong placement object name

refactor(core):
Rename onChanged functions

EdgarJRobles committed:
feature(profiles): include local parameters t-slot profile

EdgarJRobles committed:
docs(framemanager): update traslations

EdgarJRobles committed:
chore(quetzal): update quetzal workbench version

fix(fForms): include TSLOT profile

EdgarJRobles committed:
merge remote to local

EdgarJRobles committed:
fix(Quetzal_FrameBranchManager): profile path hard code

Removed profilepath variable hardcode for testing

EdgarJRobles committed:
chore(quetzal version): Update fix version to 1.6.8

EdgarJRobles committed:
perf(frameBranch): reduce trim & extend function time

EdgarJRobles committed:
dcore(readme): Sync quetzal version

petermaginot committed:
Correct wall thickness (were previously twice as thick as specified in ASME B36.10) and add Schedule 5 table

petermaginot committed:
Basically functional process to insert tees. Still a few bugs around reversing and rotating but I'm working on that.

petermaginot committed:
Fixed Tee insertion function and created new alignTwoPorts function which will connect any two pipe objects at the indicated ports

petermaginot committed:
Reverted to DN for all pipe schedules, tee function mostly working (except Reverse)

petermaginot committed:
Fixed the tee reverse function on the tee insertion form

EdgarJRobles committed:
refactor(fFeatures): rename function events and normal

EdgarJRobles committed:
Merge pull request #24 from petermaginot/master add tee fttings option

Correct wall thickness tables and create tee insertion function

petermaginot committed:
Bunch of changes:
-Added a port direction property for each port of each object, which lets you much more reliably connect two ports of objects in the correct orientation. The alignTwoPorts function now uses this property by default to align the ports on two pipe objects.
--Note that I didn't add this for the Terminal Adapter or U-bolt objects, as the terminal adapter object doesn't have ports (not sure why) and I can't really tell what the U-bolt object's port property is supposed to reflect.
-Added a doReduct function so that the reducer creation process mirrors that of all of the other pipe objects, and moved the placement logic from the pForms file to the doReduct function in pCmd. Additionally for reducers, added radio buttons to the insertion form so you can select whether you want the reducer to be connected on the large or small end.
-For caps, added the alignTwoPorts function call to reliably connect caps in the proper orientation

petermaginot committed:
Merge branch 'EdgarJRobles:master' into master

BigIskander committed:
Fix UI elements overlaping issue.

EdgarJRobles committed:
Merge pull request #26 from BigIskander/master

Fix UI elements overlaping issue.

petermaginot committed:
Merge branch 'EdgarJRobles:master' into master

petermaginot committed:
-Improved selection port attachment to address inconsistency noted in Issue number 14 by adding getAttachmentPoints and getSelectionPortAttachment functions to automatically find the closest port to the selected face, edge, or vertex.
--Changed only for Pipe, Tees, Caps, and Reducers. Ells have some trickier logic that I need to understand before changing
-Changed insertion logic to only insert a single component to the first selected item, rather than looping through all selected edges/objects and inserting components on all of them. This should lead to fewer unintentional insertions.

petermaginot committed:
-Updated alignTwoPorts function to better handle case where objects are initially closely aligned or anti-aligned to keep them from blasting off into space
-Updated flange insertion logic to use alignTwoPorts to more reliably connect to existing piping components
-Updated Reverse function for reducer, flange on insertion form
-Corrected "remove pipe equivalent length" for flange to remove raised face thickness

petermaginot committed:
-Changed elbow reverse function to match other fittings
-Changed doElbow function's logic when only a single object is selected to use getAttachmentPorts function to more reliably connect correctly. I left the makeElbowBetweenThings path alone.

petermaginot committed:
-Added radio buttons on Flange insertion form to allow flange to be inserted on raised face or weld end
-Added port directions to Valve class

petermaginot committed:
Added Gasket object class

petermaginot committed:
Update gasket icon

EdgarJRobles committed:
fix(PipeBrach): activaded signal wrong implementation

EdgarJRobles committed:
Merge pull request #27 from petermaginot/master

Component insertion improvements

EdgarJRobles committed:
fix(insertpypebranch):None object condition no detected

EdgarJRobles committed:
fix(elbowForm): Proxy not instancied \r\n chore(release): update version

petermaginot committed:
-Added Beam object that can be inserted similar to the Pipe object (Beam class in fFeatures)
-Added reinforced outlet fittings (Outlet class)
-Improved appearance of Tee objects by adding appropriate fillets for reducing tees and a new construction for straight tees that more accurately represents the shape of forged butt weld tees

petermaginot committed:
Added Blind flange type

petermaginot committed:
Added NPS/DN selection toggle on all component forms, as well as default preference selection in Preferences
Improved outlet dimensional accuracy

petermaginot committed:
Add eccentric reducer rotation

petermaginot committed:
Added Socket Weld/Threaded Elbows

petermaginot committed:
Add socket weld tees

petermaginot committed:
Added Socket weld/threaded caps

EdgarJRobles committed:
docs(README): add roadmap for quetzal workbench

petermaginot committed:
Added socket couplings and unions

petermaginot committed:
Improved appearance of Outlet fittings by adding fishmouth cut of carrier pipe OD

petermaginot committed:
Pass rating to all piping component classes rather than hard coding.
Add icon for Coupling/Union
Add valve property tables for additional valve types (not yet implemented)

petermaginot committed:
Added socket/threaded valve

EdgarJRobles committed:
Merge branch 'master' into master

EdgarJRobles committed:
Merge pull request #29 from petermaginot/master

Numerous component additions/improvements and DN/NPS selector improvement

petermaginot committed:
Merge branch 'master' of https://github.com/petermaginot/quetzal

petermaginot committed:
Add flanged valves, with either handle or gearbox actuator

petermaginot committed:
-Added option to insert bolts and nuts along with gasket on Gasket insertion form
-Corrected a 1 degree rotation bug when flanges are inserted which caused bolt hole misalignment
-Corrected flange thickness for weld neck and slip on flanges

petermaginot committed:
Add bolt CSV tables

petermaginot committed:
Added flange schedule selection for weld neck flanges, which avoids the need for numerous property tables for various schedule/class combinations

petermaginot committed:
Fix flange rating pass bug

petermaginot committed:
-Fix bug with component insertion forms when the item ended in the letters "C" "S" or "V".
-Improve Branch icon to accurately represent how it looks.

petermaginot committed:
Fixed issue with elbow placement on PypelineManager insertion
Corrected butt weld elbow schedule tables with standard elbow sizes and schedueles
Created new FreeCAD Wiki page for Quetzal and added link

petermaginot committed:
Modify BreakPipe function to allow any unit input

petermaginot committed:
Updated the Mate Edges tool to use pipe object's ports when available instead of just matching edges.

petermaginot committed:
Improved tooltips for several components

petermaginot committed:
Add "Remove pipe equivalent length" check boxes to the Tee and Elbow insertion forms which removes pipe length to fitting center, similar to the check box already present on the flange insertion form.

petermaginot committed:
fix comment

petermaginot committed:
Added PSize2 property to Reduct class objects to make size matching more reliable

petermaginot committed:
Improve automatic size matching among components. For objects where it is possible, when the insertion form is loaded the code will attempt to automatically match the pipe schedule and size. For objects where automatic schedule matching isn't possible or practical (fittings with socket weld options and flanges) the code will autmatically match and select the matching pipe size once a schedule or rating selection is made.

petermaginot committed:
Made rotation angles directly editable for elbows, tees, eccentric reducers, and valves

EdgarJRobles committed:
fix(FrameForm): reversed mapping force \n feature(FrameForm): Add reverse mapping memory \n feature(dododialogs): add preview images

petermaginot committed:
Merge update to insertion form and updated the new objects to use the new style of insertion form. I also made some changes to the preview image generator, so that files are created with only the DN size, not the NPS size if it is selected. It could still use some testing to make sure it works for all cases.

petermaginot committed:
Fix typo and block error that occus when save file loaded.

petermaginot committed:
Fix quantity parser variable back to size_selected to match master

petermaginot committed:
Missed a few parse quantities with old format

petermaginot committed:
The number of disconnects from the parent class was getting ridiculous so move it back to the dodoDialogs.py document. Also fix reducer CSV tables.

petermaginot committed:
-Fix CSV file encoding issue with some files and fix 150lb blind flange csv file
-Remove automatic grade selection from components with socket weld and butt weld options.

petermaginot committed:
Improve preview image generation reliability, always hide other objects while generating preview.
Fix bug with some reducing tees to render properly
Fix bug on XXH reducer tables
Add images for all objects currently accessible
Added macro for mass preview image generation to the folder

petermaginot committed:
Fix comment

petermaginot committed:
Fix bug where preview image would load before selection was made, showing incorrect item

EdgarJRobles committed:
fix merge conflicts

EdgarJRobles committed:
Merge branch 'petermaginot-master'

EdgarJRobles committed:
feat(dodoDialogs): Add get pipe line function

EdgarJRobles committed:
fix(pForms): ComboBox for select existing object with wrong name

EdgarJRobles committed:
refac(dodoDialogs): remove duplicate rate label & remove extra spaces

Remove Rate: label due duplicate information on a existing combobox

EdgarJRobles committed:
feat(fFeatures): Add bisec plane generation on wires

Add support for wire base lines, refactor generate bisec planes to meka
it modular

petermaginot committed:
Correct bug with "Remove equivalent pipe length" check box for slip-on flanges. Flange face now ends up exactly where edge of pipe was now.
As a side note, I've noticed a more general bug with the "Remove equivalent pipe length" check box for all fittings when the pipe length is less than 2x the amount that needs to be trimmed. Not fixed here but I'll be loking at this next.

petermaginot committed:
Fix bug where fCmd.extendTheBeam fails to correctly trim pipe when the trim length is more than 1/2 the pipe length by creating a new dedicated pipe-trimming function, shortenPipeAtPort, that trims a pipe from a specified port. This fix was applied to doElbow single elbow insertion path (left as-is for the insertElbowBetweenThings path), doSocketElbow, doTee, doSocketTee, and doFlanges, which are all of the components that currently have a "trim pipe" option.
While troubleshooting, I noticed an additional bug that sometimes caused a spurious component insertion at the origin caused by the try/except error catching being triggered after a component had already been inserted, causing a spurious additional component insertion at the origin.  I changed it for all of these components to an if/else path to match the existing doElbow logic. I didn't change any of the other components, but this is something to keep an eye on if there are any other cases that can cause this bug to arise. It was being caused by a lack of a self.execute call in the __init__ function for the tee class in pFeatures so I added a self.execute line for all of the classes that lacked one in pFeatures.

EdgarJRobles committed:
Merge pull request #34 from petermaginot/master

Fix pipe trimming bug noted in issue #33

EdgarJRobles committed:
feat(FrameBranch): Allow FrameBranch object to toggle Visibility

Toggle internal objects Visibility

EdgarJRobles committed:
fix(addBeams): fix nest new beam into existing frameBranch object

EdgarJRobles committed:
fix(trimOrExtend): fix AttachmentOffset direction

EdgarJRobles committed:
Merge branch 'devel' of github.com:EdgarJRobles/quetzal into devel

gitgrahamdunn committed:
Add check valve support

gitgrahamdunn committed:
Add gate valve support

gitgrahamdunn committed:
Add plug valve support

gitgrahamdunn committed:
Add floating and trunnion ball valve support

gitgrahamdunn committed:
Add flanged swing check valves

gitgrahamdunn committed:
Fix duplicate SW fitting rows and complete 3000lb socket union sizes

gitgrahamdunn committed:
Add 6000 and 9000 socket fitting tables

gitgrahamdunn committed:
Add XXS butt-weld tee table

gitgrahamdunn committed:
Add knife gate valve support

EdgarJRobles committed:
Merge pull request #37 from gitgrahamdunn/add-check-valve-support

Add check valve support

gitgrahamdunn committed:
Add pinch valve support

gitgrahamdunn committed:
Add beam clamp support

EdgarJRobles committed:
Merge branch 'master' into add-gate-valve-support

EdgarJRobles committed:
Merge pull request #38 from gitgrahamdunn/add-gate-valve-support

Add gate valve support

gitgrahamdunn committed:
Add rectangular HVAC duct elbows

gitgrahamdunn committed:
Add rectangular HVAC duct reductions

gitgrahamdunn committed:
Add catalog-backed pipe reducer schedules

EdgarJRobles committed:
feat(dodoDialogs): add rotation on valves

this rotation allow best perspective view for screenshot

EdgarJRobles committed:
Merge branch 'master' into add-ball-valve-support

EdgarJRobles committed:
Merge pull request #40 from gitgrahamdunn/add-ball-valve-support

Add floating and trunnion ball valve support

EdgarJRobles committed:
Merge branch 'master' into add-plug-valve-support

EdgarJRobles committed:
Merge pull request #39 from gitgrahamdunn/add-plug-valve-support

Add plug valve support

EdgarJRobles committed:
Merge branch 'master' into add-swing-check-valve-support

EdgarJRobles committed:
Merge pull request #41 from gitgrahamdunn/add-swing-check-valve-support

Add flanged swing check valves

EdgarJRobles committed:
Merge branch 'devel'

feat(dodoDialogs): add rotation on valves
fix(trimOrExtend): fix AttachmentOffset direction
fix(addBeams): fix nest new beam into existing frameBranch object
feat(FrameBranch): Allow FrameBranch object to toggle Visibility

EdgarJRobles committed:
Merge pull request #44 from gitgrahamdunn/add-socket-fittings-quality

Expand socket-weld fitting data

EdgarJRobles committed:
Merge branch 'master' into add-knife-gate-valves

EdgarJRobles committed:
Merge pull request #46 from gitgrahamdunn/add-knife-gate-valves

Add knife gate valve support

EdgarJRobles committed:
Merge pull request #45 from gitgrahamdunn/add-xxs-buttweld-tees

Add XXS butt-weld tee data

EdgarJRobles committed:
Merge branch 'master' into add-pinch-valves

EdgarJRobles committed:
Merge pull request #47 from gitgrahamdunn/add-pinch-valves

Add pinch valve support

EdgarJRobles committed:
fix(dodoPM): Modify prop dict to allow build objects

Roy-043 committed:
Remove call to init_toolbar from InitGui.py

The init_toolbar function has been removed from the Draft code.

Commit:
fix: make the geometry builders usable without a GUI

Two independent defects stopped pFeatures/pCmd being driven from a headless
process. Both are small; neither touches geometry.

1. NameError in Valve._execute_legacy

   `rating` was read but never bound, so constructing a butterfly, globe,
   wafer-check or generic ball valve raised NameError before it could build a
   shape -- 30 catalogue rows dead. Introduced by the "Add gate valve support"
   change, which replaced `fp.PRating.find(...)` with a bare `rating.find(...)`
   without binding the local. Knife gate and pinch survived only because they
   early-return above the fault.

   Fixed by binding `rating = fp.PRating.lower()` before use, which also makes
   the comparison case-insensitive as the surrounding early-returns already are.

2. ViewProvider assumed a GUI

   In console mode DocumentObject.ViewObject is None -- FreeCAD returns
   Py::None when FreeCADGui holds no Gui::Document for the App document -- so
   `obj.Proxy = self` raised AttributeError and every make*() factory died
   before returning. Guarding once inside ViewProvider covers all 17 call sites
   and any added later. Two direct ViewObject attribute writes (makeTank's
   colour/transparency, makePypeLine2's provider) are guarded the same way, on
   the pattern fFeatures.py already uses for Draft.

   The geometry itself never needed the GUI: pFeatures is pure Part/BRep.

Verified under FreeCAD 1.1.0 with QT_QPA_PLATFORM=offscreen, FreeCAD.GuiUp=0:
butterfly/globe/wafer-check/ball now build valid solids, and pCmd.makeElbow
returns a valid 4-face shell with both ports.

Commit:
fix: Flange builds a null shape and Tee crashes in its own error handler

Both only bite without a GUI, which is why they have gone unnoticed.

Flange.execute sets fp.ViewObject.Deviation unconditionally. ViewObject is None
in console mode, the resulting AttributeError is swallowed by recompute, and
every Flange comes back with a NULL shape and no error to explain it.
Tessellation quality is a display setting, so it is now set only when there
is a GUI to set it on.

Tee's fillet fallback crashes inside itself. fillet_r is a Base.Quantity and
"{:.2f}".format(Quantity) raises TypeError, so the handler written to keep a
failed fillet from taking down the recompute is the thing taking down the
recompute. float() at the format site.

Also: the flange neck is built from ODp/T1 even when both are zero. They are
documented OPTIONAL, and Part.makeCylinder(0, 0, ...) fuses a degenerate solid
that leaves the flange isValid() False. Built only when the dimensions are real.

Reproduced under FreeCAD 1.1.0 with QT_QPA_PLATFORM=offscreen: before, Flange
returns a null shape and Tee raises TypeError; after, both build valid solids.

EdgarJRobles committed:
Add link to 2D nomenclature drawings

EdgarJRobles committed:
refactor(tablez): add extra elbow,pipe,Cap CVS tablez

Thanks to vorteccpi@carolina.rr.com Paul Bercker
and also added a 2d drawings markdown files to explain CVS estructure
and variables

EdgarJRobles committed:
Merge branch 'master' of github.com:EdgarJRobles/quetzal

merge missing changes from remote master

EdgarJRobles committed:
Change image paths to relative in Nomeclature.md

Updated image paths in the Nomeclature.md file to use relative paths instead of absolute paths.

EdgarJRobles committed:
Merge pull request #57 from Roy-043/Remove-call-to-init_toolbar-from-InitGui.py

Remove call to init_toolbar from InitGui.py

EdgarJRobles committed:
Merge pull request #58 from puran-water/fix/headless-geometry

Fix three defects that stop pFeatures being used without a GUI

EdgarJRobles committed:
Update bump version 1.8.10 package.xml

EdgarJRobles committed:
update dump version to 1.8.10 quetzal_config.py

EdgarJRobles committed:
fix(pForms): currentRatingLab not existing any more

-Remove old qt form variable from previous dialog versions
-Added extra CVS files for Tee category thank again to Paul Becker vorteccpi@carolina.rr.com

EdgarJRobles committed:
Merge pull request #48 from gitgrahamdunn/add-beam-clamps

Add beam clamp support

EdgarJRobles committed:
Merge branch 'master' into add-hvac-duct-elbows

EdgarJRobles committed:
Merge pull request #50 from gitgrahamdunn/add-hvac-duct-elbows

Add rectangular HVAC duct elbows

EdgarJRobles committed:
Merge branch 'master' into add-hvac-duct-reductions

EdgarJRobles committed:
Merge pull request #51 from gitgrahamdunn/add-hvac-duct-reductions

Add rectangular HVAC duct reductions

EdgarJRobles committed:
Merge pull request #52 from gitgrahamdunn/add-catalog-pipe-reducers

Add catalog-backed pipe reducer schedules

EdgarJRobles committed:
fix(fFeatures): Local profiles did not get csv properties

only BIM profiles were extrated during profile selection
